repair/repair.cc

```diff
@@ -1204,7 +1222,13 @@ future<> repair::user_requested_repair_task_impl::run() {
         }
 
         bool hints_batchlog_flushed = false;
-        auto participants = get_hosts_participating_in_repair(germs->get(), keyspace, ranges, data_centers, hosts, ignore_nodes).get();
+        std::list<gms::inet_address> participants;
+        if (_small_table_optimization) {
+            auto normal_nodes = germs->get().get_token_metadata().get_all_endpoints();
+            participants = std::list<gms::inet_address>(normal_nodes.begin(), normal_nodes.end());
```

Comment:
@asias doesn't the row-level repair algorithm assume that all passed-in peers own the currently repaired range? If so, won't it attempt to replicate all data to all nodes, even if said node doesn't own some of it?

Reply:
Yes.